metricbeat/module/elasticsearch/elasticsearch.go

```diff
@@ -0,0 +1,129 @@
+package elasticsearch
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+
+	"github.com/elastic/beats/metricbeat/helper"
+)
+
+// Global clusterIdCache. Assumption is that the same node id never can belong to a different cluster id
+var clusterIDCache = map[string]string{}
+
+// Info construct contains the data from the Elasticsearch / endpoint
+type Info struct {
+	ClusterName string `json:"cluster_name"`
+	ClusterID   string `json:"cluster_uuid"`
+}
+
+// GetClusterID fetches cluster id for given nodeID
+func GetClusterID(http *helper.HTTP, uri string, nodeID string) (string, error) {
+
+	// Check if cluster id already cached. If yes, return it.
+	if clusterID, ok := clusterIDCache[nodeID]; ok {
+		return clusterID, nil
+	}
+
+	// Makes sure the http uri is reset to its inital value
+	defer http.SetURI(uri)
+
+	info, err := GetInfo(http, uri)
+	if err != nil {
+		return "", err
+	}
+
+	clusterIDCache[nodeID] = info.ClusterID
+	return info.ClusterID, nil
+}
+
+// IsMaster checks if the given node host is a master node
+//
+// The detection of the master is done in two steps:
+// * Fetch node name from /_nodes/_local/name
+// * Fetch current master name from cluster state /_cluster/state/master_node
+//
+// The two names are compared
+func IsMaster(http *helper.HTTP, uri string) (bool, error) {
+	// Makes sure the http uri is reset to its inital value
+	defer http.SetURI(uri)
+
+	node, err := getNodeName(http, uri)
+	if err != nil {
+		return false, err
+	}
+
+	master, err := getMasterName(http, uri)
+	if err != nil {
+		return false, err
+	}
+
+	return master == node, nil
+}
+
+func getNodeName(http *helper.HTTP, uri string) (string, error) {
+
```

Comment:
There are a few methods with a starting empty line, it would be nice to remove them

Reply:
removed all of them.